lib/libc/arcmwdt/threading.c

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright (c) 2021 Synopsys.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifdef CONFIG_MULTITHREADING
+
+#include <init.h>
+#include <kernel.h>
+#include <sys/__assert.h>
+#include <sys/mutex.h>
+#include <../lib/src/c/inc/internal/thread.h>
+
+void _mwmutex_create(_lock_t *mutex_ptr)
+{
+#ifndef CONFIG_USERSPACE
+	*mutex_ptr = k_malloc(sizeof(struct k_mutex));
+#else
+	*mutex_ptr = k_object_alloc(K_OBJ_MUTEX);
+#endif /* !CONFIG_USERSPACE */
+	__ASSERT(*mutex_ptr != NULL, "MWDT lock allocation failed");
```

Comment:
since __mwmutex_create doesn't have a return value to propagate errors, it would be better to always check and fail instead of only when asserts are enabled.

Reply:
Now we call k_panic instead in that case.